linalg/superlu.cpp

```diff
@@ -163,6 +163,158 @@ SuperLURowLocMatrix::SuperLURowLocMatrix( const HypreParMatrix & hypParMat )
                                   SLU_NR_loc, SLU_D, SLU_GE);
 }
 
+SuperLURowLocMatrix::SuperLURowLocMatrix( const BlockOperator &ops )
+   : rowLocPtr_(NULL)
+{
+
+   rowLocPtr_      = new SuperMatrix;
+   SuperMatrix * A = (SuperMatrix*)rowLocPtr_;
+
+   A->Store = NULL;
+
+   comm_ = (dynamic_cast<HypreParMatrix&>(ops.GetBlock(0,0))).GetComm();
+
+   int myid, num_procs;
+   MPI_Comm_rank(comm_, &myid);
+   MPI_Comm_size(comm_,&num_procs);
+
+   // First cast the parameter to a hypre_ParCSRMatrix
+   int numsysrows = ops.NumRowBlocks();
+   int numsyscols = ops.NumColBlocks();
+
+   Array2D<hypre_ParCSRMatrix *> parcsrmats;
+   Array2D<hypre_CSRMatrix *> mats;
+   parcsrmats.SetSize(numsysrows, numsyscols);
+   mats.SetSize(numsysrows, numsyscols);
+
+   int nnz = 0;
+   int global_cols = 0;
+   int global_rows = 0;
+   int local_rows = 0;
+   int local_cols = 0;
+   int first_index = 0;
+
+   // Fill csr arrays
+   for (int i=0; i<numsysrows; i++)
+   {
+      for (int j=0; j<numsyscols; j++)
+      {
+         if (!ops.IsZeroBlock(i,j))
+         {
+            Operator &op = ops.GetBlock(i,j);
+            parcsrmats(i,j) = (hypre_ParCSRMatrix *)dynamic_cast<HypreParMatrix&>(op);
```

Comment:
This dynamic cast looks a little fishy casting from HypreParMatrix and promoting to ParCSR, what if it's not a ParCSR?

Reply:
That's a good question. I was honestly just following the pattern of the [`SuperLURowLocMatrix(const HypreParMatrix & hypParMat)`](https://github.com/mfem/mfem/blob/a6f1582bf8809cefa0d3676c8cbf51e2172cbbab/linalg/superlu.cpp#L124) constructor. I'll take a closer look at this.
- [ ] Examine ParCSR casting issue